presto-main/src/main/java/com/facebook/presto/security/AccessControl.java

```diff
@@ -37,6 +37,11 @@
     Set<String> filterCatalogs(Identity identity, Set<String> catalogs);
 
     /**
+     * Check whether identity is allowed to access catalog
+     */
+    void checkCanAccessCatalog(Identity identity, String catalogName);
```

Comment:
All the checkCanAccessCatalog changes belong in https://github.com/prestodb/presto/pull/7695/commits/30730672acfcc5e73d4456ae05ac6394a1a7ec73, this commit should just be adding the file based system access control.